extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java

```diff
@@ -139,5 +313,236 @@ public void registerSwaggerUiServletExtension(SwaggerUiRecorder recorder,
+
+        /**
+         * Controls the display of operationId in operations list. The default is false.
+         */
+        @ConfigItem(name = "displayOperationId")
+        Optional<Boolean> displayOperationId;
+
+        /**
+         * The default expansion depth for models (set to -1 completely hide the models).
+         */
+        @ConfigItem(name = "defaultModelsExpandDepth")
+        Optional<Integer> defaultModelsExpandDepth;
+
+        /**
+         * The default expansion depth for the model on the model-example section.
+         */
+        @ConfigItem(name = "defaultModelExpandDepth")
+        Optional<Integer> defaultModelExpandDepth;
+
+        /**
+         * Controls how the model is shown when the API is first rendered.
+         */
+        @ConfigItem(name = "defaultModelRendering")
+        Optional<String> defaultModelRendering;
+
+        /**
+         * Controls the display of the request duration (in milliseconds) for "Try it out" requests.
+         */
+        @ConfigItem(name = "displayRequestDuration")
+        Optional<Boolean> displayRequestDuration;
+
+        /**
+         * Controls the default expansion setting for the operations and tags.
+         */
+        @ConfigItem(name = "docExpansion")
+        Optional<DocExpansion> docExpansion;
+
+        /**
+         * If set, enables filtering. The top bar will show an edit box that you can use to filter the tagged operations that
+         * are shown.
+         * Can be Boolean to enable or disable, or a string, in which case filtering will be enabled using that string as the
+         * filter expression.
+         * Filtering is case sensitive matching the filter expression anywhere inside the tag.
+         */
+        @ConfigItem
+        Optional<String> filter;
+
+        /**
+         * If set, limits the number of tagged operations displayed to at most this many. The default is to show all operations.
+         */
+        @ConfigItem(name = "maxDisplayedTags")
+        Optional<Integer> maxDisplayedTags;
+
+        /**
+         * Apply a sort to the operation list of each API.
+         * It can be 'alpha' (sort by paths alphanumerically), 'method' (sort by HTTP method) or a function (see
+         * Array.prototype.sort() to know how sort function works).
+         * Default is the order returned by the server unchanged.
+         */
+        @ConfigItem(name = "operationsSorter")
+        Optional<String> operationsSorter;
+
+        /**
+         * Controls the display of vendor extension (x-) fields and values for Operations, Parameters, and Schema.
+         */
+        @ConfigItem(name = "showExtensions")
+        Optional<Boolean> showExtensions;
+
+        /**
+         * Controls the display of extensions (pattern, maxLength, minLength, maximum, minimum) fields and values for
+         * Parameters.
+         */
+        @ConfigItem(name = "showCommonExtensions")
+        Optional<Boolean> showCommonExtensions;
```

Comment:
I think all these `Optional<Boolean>` should be made `boolean` with the proper default value. You just need to define `defaultValue="true"` if the default value is true. False is the default.

Reply:
I did not want to do that, as it will make the html unnecessary bigger, the default is actually nothing, i.e fall back to the default as in swagger ui.